### PR TITLE
[api-minor] Use `new Worker()` syntax in webpack entrypoint

### LINKED ISSUE #15426: Update `pdfjs-dist/webpack` entrypoint to use `new Worker()` instead of `worker-loader`
In Webpack 5, the [recommended way](https://webpack.js.org/guides/web-workers/) to create a web worker is

```ts
new Worker(new URL('./worker.js', import.meta.url));
```

Could we update\* the [`pdfjs-dist/webpack` entrypoint](https://github.com/mozilla/pdfjs-dist/blob/master/webpack.js) to use this syntax instead of the legacy `worker-loader`? One of the Webpack contributors told me that `worker-loader` uses "magic" and can cause issues.

\* To maintain compatibility with people still using Webpack 4, we should probably introduce a new entrypoint called `pdfjs-dist/webpack5` (or something like that) rather than changing the existing entrypoint.

Can attempt a PR if you would like.

Closely-related: https://github.com/mozilla/pdf.js/issues/15302

### PR DESCRIPTION
**Pull request by srmagura**
Closes #15426.

**This requires Webpack 5 and will break for anyone using Webpack 4.** `worker-loader` no longer needs to be installed.

I tested this change in my project and it worked.

### CODE REVIEW

File: examples/webpack/README.md [hunk 1]
```diff
@@ -21,17 +21,14 @@ the worker code, and the `workerSrc` path shall be set to the latter file.
 ### Minification
 
 If you are configuring Webpack to output a minified build, please note that you
-*must* configure the minifier to keep original class/function names intact;
+_must_ configure the minifier to keep original class/function names intact;
```
Comment (Snuffleupagus) [inline examples/webpack/README.md:24]: This looks like an unrelated change, since the word is already italicized when looking at https://github.com/mozilla/pdf.js/tree/master/examples/webpack#minification; please revert this change.

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
Please also remove the `worker-loader` related code [from `gulpfile.js`](https://github.com/mozilla/pdf.js/blob/493bb65005202678c5fa803859dcf70cbb6e37a6/gulpfile.js#L2174-L2181).

Furthermore, please note that we don't tag PRs (or commit messages) with "Breaking change" in this repository. Instead we use "[api-minor]" (or "[api-major]") for that purpose, so please update the commit message accordingly.

When updating the patch, please note https://github.com/mozilla/pdf.js/wiki/Squashing-Commits

File: examples/webpack/README.md (hunk 1 above)
Comment (srmagura) [reply examples/webpack/README.md:24]: This change was made automatically by my Prettier extension. If I revert the change, the next person who edits the file may encounter the same issue. Do you still want me to revert it?

**Comment by srmagura** [comment]
Thank you, I'm not sure how I missed that `worker-loader` was referenced in the gulpfile.

File: examples/webpack/README.md (hunk 1 above)
Comment (Snuffleupagus) [reply examples/webpack/README.md:24]: As a general rule, we *try* to avoid making unrelated changes in patches.

> This change was made automatically by my Prettier extension.

Sorry, but considering that we're not using such a thing in this project that's maybe not entirely relevant here.

If the suggestion is that we should start using such a plugin, then that's a discussion that ought to happen elsewhere.
(For that sort of changes to be considered though, such a task *should* run as part of `gulp lint` and we'd also want to change *all* README files at once rather than piecemeal like this.)

> Do you still want me to revert it?

Yes, otherwise I'd not have comment on this :-)

**Review by Snuffleupagus — APPROVED** [review]
Thank you!
